### PR TITLE
Moved DES to AES migration outside of GoCipher

### PR DESCRIPTION
**Pull request by maheshp**
* Removing support for DES to AES as part of post construct
* Moved DES to AES migration outside of GoCipher

### CODE REVIEW

File: config/config-api/src/main/java/com/thoughtworks/go/domain/config/ConfigurationProperty.java [hunk 1]
```diff
@@ -266,14 +266,6 @@ public void initialize() {
         if (configurationValue == null && encryptedValue == null) {
             configurationValue = new ConfigurationValue("");
         }
-
-        if (encryptedValue != null && isNotBlank(encryptedValue.getValue())) {
-            setEncryptedValue(cipher.maybeReEncryptForPostConstructWithoutExceptions(encryptedValue.getValue()));
-        }
-    }
-
-    private void setEncryptedValue(String encryptedValue) {
```
Comment (ketan) [inline config/config-api/src/main/java/com/thoughtworks/go/domain/config/ConfigurationProperty.java]: Leave this be. Might be used from within rails.

Comment (ketan) [reply config/config-api/src/main/java/com/thoughtworks/go/domain/config/ConfigurationProperty.java]: Make it a public method, perhaps.

File: config/config-api/src/test/java/com/thoughtworks/go/security/DesToAesMigratorTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.spy;
+
+public class DesToAesMigratorTest {
```
Comment (ketan) [inline config/config-api/src/test/java/com/thoughtworks/go/security/DesToAesMigratorTest.java]: Use junit5?

Comment (maheshp) [reply config/config-api/src/test/java/com/thoughtworks/go/security/DesToAesMigratorTest.java]: Done

File: config/config-api/src/main/java/com/thoughtworks/go/domain/config/ConfigurationProperty.java (hunk 1 above)
Comment (maheshp) [reply config/config-api/src/main/java/com/thoughtworks/go/domain/config/ConfigurationProperty.java]: Got it back.

**Pull request merged**